File: src/main/java/com/samla/sdk/analytics/Analytics.java
package com.samla.sdk.analytics;

import android.content.Context;

import com.google.firebase.analytics.FirebaseAnalytics;

public class Analytics {

    private static Context applicationContext;

    public Analytics(Context context) {
        applicationContext = context;
    }

   public static FirebaseAnalytics getFirebaseAnalytics () {
        return FirebaseAnalytics.getInstance(applicationContext);
   }

}

File: src/main/java/com/samla/sdk/userflow/Analytics.kt
package com.samla.sdk.userflow

import android.content.Context
import com.google.firebase.analytics.FirebaseAnalytics
import com.samla.sdk.ISamla
import com.samla.sdk.userflow.funnel.FunnelManager

object Analytics {

    lateinit var ISamla: ISamla

    fun setClientActivity(entry : ISamla) {
        ISamla = entry;
    }

    fun getFirebaseAnalytics (): FirebaseAnalytics {
        return FirebaseAnalytics.getInstance(ISamla.getActivity())
    }

}